zenoh/src/api/handlers/ring.rs

```diff
@@ -62,7 +67,29 @@ impl<T> RingChannelHandler<T> {
         }
     }
 
-    /// Receive from the ring channel. If the ring channel is empty, this call will block until an element is available in the channel.
+    /// Receive from the ring channel with a deadline.
+    ///
+    /// If the ring channel is empty, this call will block until an element is available in the channel,
+    /// or return `None` if the deadline has passed.
+    pub fn recv_deadline(&self, deadline: Instant) -> ZResult<Option<T>> {
+        let Some(channel) = self.ring.upgrade() else {
+            bail!("The ringbuffer has been deleted.");
+        };
+
+        loop {
```

Comment:
Why the loop?

Reply:
Because of half-backed PR 😓  it should be better now.

Reply:
I mean, there is a loop because there is one in `recv`, but I forgot to catch the timeout error.